pkg/restapi/edv/operation/operations.go

```diff
@@ -244,51 +256,105 @@ func (vc *VaultCollection) createDocument(vaultID string, document EncryptedDocu
 		return err
 	}
 
+	// TODO: Index creation and document storage should all happen in a single transaction (all-or-nothing).
+	// If one of these requests fails, then the database will be left in a weird state.
+	// https://github.com/trustbloc/edv/issues/49
+
+	// TODO: Don't recreate an index if it already exists.
+	// https://github.com/trustbloc/edge-core/issues/26 and https://github.com/trustbloc/edv/issues/50
```

Comment:
I think this should be done after this PR is merged in. What do you think? @bstasyszyn @fqutishat

(this will require changes to edge-core)